### DISCUSSION TITLE
Was there a reason for IP6 exclusion?

### DISCUSSION THREAD

**Discussion by Orion120** [chadgeary/cloudblock#13] [Q&A] [answered]
Total lurkr noob. Why is ip6 not addressed and where would I look to add?

**Comment by chadgeary** [comment, marked as answer]
Good question! IPv6 would be a fine addition. As far as where to look - it
would require touches on nearly every component.

For the cloud deployments (AWS, Azure, GCP, and OCI) - the VPC networks and
security rules (*-network.tf and *-security.tf) would need to define /
configure IPv6 networks/subnets, associated routing and firewall rules. GCP
doesn't support IPv6 without a load balancer though - azure only added it
as of mid this year (2020).

For cloud and standalone deployments, an IPv6 docker network needs to be
created and attached to each of the containers. I'm curious how wireguard
handles peer configurations with multiple addresses.

Can't stress this enough: GOOD IDEA AND THANK YOU

With all of the above said, in my experience IPv6-only ISPs (often mobile
carriers) will still work with wireguard as they run IPv6 to IPv4
translation.

**Comment by neitsab** [comment]
There is to me a privacy issue with not supporting IPv6: my home IPv6 leaks to any website despite my IPv4 correctly reflecting the (cloud) server's one. I'm with Vodafone in Germany, and they mainly attribute an IPv6 block to residential customer while tacking on it an IPv4 at some point to do dual-stack.

I configured Cloudblock to route (I thought) all my traffic to escape the prying eyes (and logs) of ISPs, but due to IPv6 exclusion this proved to be... Inadequate 😅

I'm still grateful for the project, but I think the lack of IPv6 support should be displayed more prominently in the README 🙂
Cheers

**Reply by neitsab** [reply]
On that note, I just confirmed that if I modify my client's config `[Peer]`'s `AllowedIPs` definition to include `::/0` besides the existing `0.0.0.0/0`, the IPv6 traffic is sinkholed and then lost at the server's level, which temporarily "fixes" the problem of my home IPv6 leaking 😁👍🏻

**Reply by chadgeary** [reply]
thank you for the feedback, i will be available in the coming days to look at this project more in depth and touch up as I've been away for quite some time.

Cheers!

**Reply by neitsab** [reply]
That's great to hear! I hope my recent flurry of messages didn't make you feel pressured; maintaining an open source project is hard, and I don't feel entitled to any SLA — or even answers 😁 I just wanted to contribute my findings for those that may stumble upon the same issues/gotchas 😊
